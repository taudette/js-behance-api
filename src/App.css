/* App */
.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
  margin-bottom: 20px;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

/* User Profile */
.Profile-panel, .Experiences-panel {
  text-align: left;
}

.followers, .following {
  border-bottom: 1px dotted black;
  color: #006080;
}

.Users-list {
  list-style: none;
}

.Project-info {
  text-align: left;
}

.Project-info ul {
  list-style: none;
  padding-left: 0;
}

.Search {
  padding-bottom: 25px;
}

.Project {
  height: 400px;
}
