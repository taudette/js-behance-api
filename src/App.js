import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { Provider } from 'react-redux'
import Home from './containers/Home'
import UserProfile from './containers/UserProfile'
import store, { history } from './store'
import { ConnectedRouter } from 'react-router-redux'
import { Route, Link } from 'react-router-dom'

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <div className="App">
            <header className="App-header">
              <Link to="/">
                <img src={logo} className="App-logo" alt="logo" />
                <h1 className="App-title">JS Behance API App</h1>
              </Link>
            </header>
            <main>
              <Route exact path="/" component={Home} />
              <Route exact path="/user/:userName" component={UserProfile} />
            </main>
          </div>
        </ConnectedRouter>
      </Provider>
    );
  }
}

export default App;
