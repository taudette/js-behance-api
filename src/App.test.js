import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import SearchContainer from './containers/Search'
import UserContainer from './containers/User'
import UsersContainer from './containers/Users'
import SearchComponent from './components/Search'
import UsersComponent from './components/Users'
import UserProfileComponent from './components/UserProfile'
import UserProjectsComponent from './components/UserProjects'
import FollowersComponent from './components/Followers'
import FollowingComponent from './components/Following'
import UserExperienceComponent from './components/UserExperience'
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

describe('App', () => {
  it('App renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  })
})

describe('Search container', () => {
  it('SearchContainer is connected', () => {
    const mockStore = configureStore([]);
    const store = mockStore({});
    const wrapper = shallow(<Provider store={store}>
      <SearchContainer />
    </Provider>);
    expect(wrapper.text()).toBe('<Connect(SearchContainer) />');
  })
})

describe('User container', () => {
  it('UserContainer is connected', () => {
    const mockStore = configureStore([]);
    const store = mockStore({});
    const wrapper = shallow(<Provider store={store}>
      <UserContainer />
    </Provider>);
    expect(wrapper.text()).toBe('<Connect(UserContainer) />');
  })
})

describe('Users container', () => {
  it('UsersContainer is connected', () => {
    const mockStore = configureStore([]);
    const store = mockStore({});
    const wrapper = shallow(<Provider store={store}>
      <UsersContainer />
    </Provider>);
    expect(wrapper.text()).toBe('<Connect(UsersContainer) />');
  })
})

describe('Search component', () => {
	const search = shallow(<SearchComponent />);
	it('Search component renders form correctly', () => {
		expect(search).toMatchSnapshot();
	});
})

describe('Users component', () => {
	const users = shallow(<UsersComponent />);
	it('Users component renders users list correctly', () => {
		expect(users).toMatchSnapshot();
	});
})

describe('UserProfile component', () => {
	const userProfile = shallow(<UserProfileComponent />);
	it('User profile component renders profile correctly', () => {
		expect(userProfile).toMatchSnapshot();
	});
})

describe('UserProjects component', () => {
	const userProjects = shallow(<UserProjectsComponent />);
	it('User projects component renders projects correctly', () => {
		expect(userProjects).toMatchSnapshot();
	});
})

describe('Followers component', () => {
	const followers = shallow(<FollowersComponent />);
	it('Followers component renders projects correctly', () => {
		expect(followers).toMatchSnapshot();
	});
})

describe('Following component', () => {
	const following = shallow(<FollowingComponent />);
	it('Following component renders projects correctly', () => {
		expect(following).toMatchSnapshot();
	});
})