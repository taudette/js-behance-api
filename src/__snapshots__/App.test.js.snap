// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Followers component Followers component renders projects correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FollowersComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Following component Following component renders projects correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FollowingComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Search component Search component renders form correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "row",
      "children": <Form
        bsClass="form"
        componentClass="form"
        horizontal={false}
        inline={true}
        onSubmit={undefined}
>
        <FormGroup
                bsClass="form-group"
                controlId="formBasicText"
                validationState={undefined}
        >
                <FormControl
                        bsClass="form-control"
                        componentClass="input"
                        name="input"
                        placeholder="search or browse users"
                        type="text"
                />
                <FormControlFeedback
                        bsClass="form-control-feedback"
                        bsRole="feedback"
                />
                <Button
                        active={false}
                        block={false}
                        bsClass="btn"
                        bsStyle="default"
                        disabled={false}
                        type="submit"
                >
                        Search
                </Button>
        </FormGroup>
</Form>,
      "componentClass": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "form",
        "children": <FormGroup
          bsClass="form-group"
          controlId="formBasicText"
          validationState={undefined}
>
          <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    name="input"
                    placeholder="search or browse users"
                    type="text"
          />
          <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
          />
          <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    type="submit"
          >
                    Search
          </Button>
</FormGroup>,
        "componentClass": "form",
        "horizontal": false,
        "inline": true,
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form-group",
          "children": Array [
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="input"
              placeholder="search or browse users"
              type="text"
/>,
            <FormControlFeedback
              bsClass="form-control-feedback"
              bsRole="feedback"
/>,
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              type="submit"
>
              Search
</Button>,
          ],
          "controlId": "formBasicText",
          "validationState": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control",
              "componentClass": "input",
              "name": "input",
              "placeholder": "search or browse users",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control-feedback",
              "bsRole": "feedback",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "default",
              "children": "Search",
              "disabled": false,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Search",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={true}
          onSubmit={undefined}
>
          <FormGroup
                    bsClass="form-group"
                    controlId="formBasicText"
                    validationState={undefined}
          >
                    <FormControl
                              bsClass="form-control"
                              componentClass="input"
                              name="input"
                              placeholder="search or browse users"
                              type="text"
                    />
                    <FormControlFeedback
                              bsClass="form-control-feedback"
                              bsRole="feedback"
                    />
                    <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              disabled={false}
                              type="submit"
                    >
                              Search
                    </Button>
          </FormGroup>
</Form>,
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form",
          "children": <FormGroup
            bsClass="form-group"
            controlId="formBasicText"
            validationState={undefined}
>
            <FormControl
                        bsClass="form-control"
                        componentClass="input"
                        name="input"
                        placeholder="search or browse users"
                        type="text"
            />
            <FormControlFeedback
                        bsClass="form-control-feedback"
                        bsRole="feedback"
            />
            <Button
                        active={false}
                        block={false}
                        bsClass="btn"
                        bsStyle="default"
                        disabled={false}
                        type="submit"
            >
                        Search
            </Button>
</FormGroup>,
          "componentClass": "form",
          "horizontal": false,
          "inline": true,
          "onSubmit": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": Array [
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="input"
                placeholder="search or browse users"
                type="text"
/>,
              <FormControlFeedback
                bsClass="form-control-feedback"
                bsRole="feedback"
/>,
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                type="submit"
>
                Search
</Button>,
            ],
            "controlId": "formBasicText",
            "validationState": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "componentClass": "input",
                "name": "input",
                "placeholder": "search or browse users",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control-feedback",
                "bsRole": "feedback",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "default",
                "children": "Search",
                "disabled": false,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Search",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`UserProfile component User profile component renders profile correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProfileComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "panel",
      "bsStyle": "default",
      "children": undefined,
      "className": "Profile-panel",
      "defaultExpanded": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "panel",
        "bsStyle": "default",
        "children": undefined,
        "className": "Profile-panel",
        "defaultExpanded": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`UserProjects component User projects component renders projects correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProjectsComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "panel",
      "bsStyle": "default",
      "children": Array [
        <h3>
          Projects
</h3>,
        <Row
          bsClass="row"
          componentClass="div"
>
          <p>
                    No projects are currently available
          </p>
</Row>,
      ],
      "defaultExpanded": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Projects",
        },
        "ref": null,
        "rendered": "Projects",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <p>
            No projects are currently available
</p>,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No projects are currently available",
          },
          "ref": null,
          "rendered": "No projects are currently available",
          "type": "p",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "panel",
        "bsStyle": "default",
        "children": Array [
          <h3>
            Projects
</h3>,
          <Row
            bsClass="row"
            componentClass="div"
>
            <p>
                        No projects are currently available
            </p>
</Row>,
        ],
        "defaultExpanded": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Projects",
          },
          "ref": null,
          "rendered": "Projects",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": <p>
              No projects are currently available
</p>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No projects are currently available",
            },
            "ref": null,
            "rendered": "No projects are currently available",
            "type": "p",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Users component Users component renders users list correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UsersComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "panel",
      "bsStyle": "default",
      "children": <Row
        bsClass="row"
        componentClass="div"
>
        <p>
                Sorry try a different search!
        </p>
</Row>,
      "defaultExpanded": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": Array [
          undefined,
          <p>
            Sorry try a different search!
</p>,
        ],
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sorry try a different search!",
          },
          "ref": null,
          "rendered": "Sorry try a different search!",
          "type": "p",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "panel",
        "bsStyle": "default",
        "children": <Row
          bsClass="row"
          componentClass="div"
>
          <p>
                    Sorry try a different search!
          </p>
</Row>,
        "defaultExpanded": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": Array [
            undefined,
            <p>
              Sorry try a different search!
</p>,
          ],
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sorry try a different search!",
            },
            "ref": null,
            "rendered": "Sorry try a different search!",
            "type": "p",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
